src/WebWorkersAPI/SharedWorker.res

```diff
@@ -0,0 +1,69 @@
+open ChannelMessagingAPI
+open WebWorkersAPI
+
+include EventTarget.Impl({
+  type t = sharedWorker
+})
+
+/**
+`make(string)`
+
+The SharedWorker() constructor creates a SharedWorker object that executes the 
+script at the specified URL. This script must obey the same-origin policy.
+
+```res
+let shared: sharedWorker = SharedWorker.make("sharedworker.js")
+```
+
+[Read more on MDN](https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker/)
+*/
+@new
+external make: string => sharedWorker = "SharedWorker"
+
+/**
+`make_withName(string, string)`
+
+The SharedWorker() constructor creates a SharedWorker object that executes the 
+script at the specified URL. This script must obey the same-origin policy.
+
+```res
+let shared: sharedWorker = SharedWorker.make("sharedworker.js", "name")
+```
+
+[Read more on MDN](https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker/)
+*/
+@new
+external make_withName: (string, string) => sharedWorker = "SharedWorker"
```

Comment:
I'm aware this is part of the documentation, but underscores were discouraged in a private conversation with the ReScript maintainers.
I should update the documentation here, but basically we want `makeWithName` here instead.

Example PR: https://github.com/rescript-lang/experimental-rescript-webapi/pull/91

Reply:
"The ReScript maintainers" 😁
![17536344442418062075274672950372](https://github.com/user-attachments/assets/de449ee3-2102-47c8-a32d-a73146219a4b)